File: 2021/05/04/main.go
package main

import "fmt"

func checkPossibility(nums []int) bool {
	var cnt = 0
	for i := 0; i < len(nums)-1; i++ {
		if nums[i] > nums[i+1] {
			cnt++
			if cnt > 1 || i > 0 && nums[i-1] > nums[i+1] && i+2 < len(nums) && nums[i+2] < nums[i] {
				return false
			}
		}
	}
	return true
}

func main() {
	var res = checkPossibility([]int{4, 2, 1})
	fmt.Println(res)
}

File: 2021/05/04/main.py
class Solution:
    def checkPossibility(self, nums) -> bool:
        cnt = 0
        for i in range(0, len(nums)-1):
            if nums[i] > nums[i+1]:
                cnt += 1
                if cnt > 1 or i > 0 and nums[i-1] > nums[i+1] and i + 2 < len(nums) and nums[i+2] < nums[i]:
                    return False
        return True
if __name__ == '__main__':
    print('__main__')
    nums = [4, 2, 3]
    res = Solution().checkPossibility(nums)
    print(res)
    